File: specification/specs/cat/cat_thread_pool/CatThreadPoolRecord.ts
/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

class CatThreadPoolRecord {
  /**
   * node name
   */
  'node_name'?:string
  /**
   * persistent node id
   */
  'node_id'?:NodeId
  /**
   * ephemeral node id
   */
  'ephemeral_node_id'?:string
  /**
   * process id
   */
  'pid'?:string
  /**
   * host name
   */
  'host'?:string
  /**
   * ip address
   */
  'ip'?:string
  /**
   * bound transport port
   */
  'port'?:string
  /**
   * thread pool name
   */
  'name'?:string
  /**
   * thread pool type
   * @aliases t
   */
  'type'?:string
  /**
   * number of active threads
   * @aliases a
   */
  'active'?:string
  /**
   * number of threads
   * @aliases psz
   */
  'pool_size'?:string
  /**
   * number of tasks currently in queue
   * @aliases q
   */
  'queue'?:string
  /**
   * maximum number of tasks permitted in queue
   * @aliases qs
   */
  'queue_size'?:string
  /**
   * number of rejected tasks
   * @aliases r
   */
  'rejected'?:string
  /**
   * highest number of seen active threads
   * @aliases l
   */
  'largest'?:string
  /**
   * number of completed tasks
   * @aliases c
   */
  'completed'?:string
  /**
   * core number of threads in a scaling thread pool
   * @aliases cr
   */
  'core'?:string
  /**
   * maximum number of threads in a scaling thread pool
   * @aliases mx
   */
  'max'?:string
  /**
   * number of threads in a fixed thread pool
   * @aliases sz
   */
  'size'?:string
  /**
   * thread keep alive time
   * @aliases ka
   */
  'keep_alive'?:string
}

File: output/csharp/Cat/CatThreadPool/CatThreadPoolRecord.cs
using Nest.Internal;
using Nest.CommonOptions;
using Nest.Cat;
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
namespace Nest.Cat {

	public class CatThreadPoolRecord : ICatRecord {
		
		[DataMember(Name="active")]
		public int Active { get; set; }


		[DataMember(Name="completed")]
		public long Completed { get; set; }


		[DataMember(Name="core")]
		public int Core { get; set; }


		[DataMember(Name="ephemeral_node_id")]
		public string EphemeralNodeId { get; set; }


		[DataMember(Name="host")]
		public string Host { get; set; }


		[DataMember(Name="ip")]
		public string Ip { get; set; }


		[DataMember(Name="keep_alive")]
		public Time KeepAlive { get; set; }


		[DataMember(Name="largest")]
		public int Largest { get; set; }


		[DataMember(Name="max")]
		public int Max { get; set; }


		[DataMember(Name="name")]
		public string Name { get; set; }


		[DataMember(Name="node_id")]
		public string NodeId { get; set; }


		[DataMember(Name="node_name")]
		public string NodeName { get; set; }


		[DataMember(Name="pool_size")]
		public int PoolSize { get; set; }


		[DataMember(Name="port")]
		public int Port { get; set; }


		[DataMember(Name="pid")]
		public int Pid { get; set; }


		[DataMember(Name="queue")]
		public int Queue { get; set; }


		[DataMember(Name="queue_size")]
		public int QueueSize { get; set; }


		[DataMember(Name="rejected")]
		public long Rejected { get; set; }


		[DataMember(Name="size")]
		public int Size { get; set; }


		[DataMember(Name="type")]
		public string Type { get; set; }

	}
}
